src/components/textfieldControl/js/Textfield.js

```diff
@@ -0,0 +1,138 @@
+        },
+        listeners: {
+            "onCreate.bindChangeEvt": {
+                "this": "{that}.container",
+                "method": "change",
+                "args": ["{that}.setModel"]
+            },
+            "onCreate.initTextfieldAttributes": {
+                "this": "{that}.container",
+                method: "attr",
+                args: [{
+                    "aria-labelledby": "{that}.options.ariaOptions.aria-labelledby",
+                    "aria-label": "{that}.options.strings.aria-label"
+                }]
+            }
+        },
+        invokers: {
+            setModel: {
+                changePath: "value",
+                value: "{arguments}.0.target.value"
+            }
+        }
+    });
+
+    /**
+     * Sets the model value only if the new value is a valid number, and will reset the textfield to the current model
+     * value otherwise.
+     *
+     * @param {Object} that - the component
+     * @param {Number} value - the new numerical entry
+     * @param {String} path - the path into the model for which the value should be set
+     */
+    fluid.textfield.setModelRestrictToNumbers = function (that, value, path) {
+        var isNumber = !isNaN(Number(value));
+        if (isNumber) {
+            that.applier.change(path, value);
+        }
+
+        // Set the textfield to the latest valid entry.
+        // This handles both the cases where an invalid entry was provided, as well as cases where a valid number is
+        // rounded. In the case of rounded numbers this ensures that entering a number that rounds to the current
+        // set value, doesn't leave the textfield with the unrounded number present.
+        that.container.val(that.model.value);
+    };
+
+    /******************************
+     * TextField Range Controller *
+     ******************************/
+
+    /**
+     * Range Controller is intended to be used as a subcomponent of a fluid.textfield component. It will limit the input
+     * to be constrained within a given numerical range. This should be paired with configuring the textfield.setModel
+     * invoker to use fluid.textfield.setModelRestrictToNumbers.
+     * The Range Controller is useful when combining the textfield with a UI control element such as stepper buttons
+     * or a slider to enter numerical values.
+     */
+    fluid.defaults("fluid.textfield.rangeController", {
+        gradeNames: ["fluid.textfield"],
+        components: {
+            controller: {
+                type: "fluid.modelComponent",
+                options: {
+                    model: {
+                        value: null
+                    },
+                    modelRelay: [{
+                        source: "value",
+                        target: "{fluid.textfield}.model.value",
+                        singleTransform: {
+                            type: "fluid.transforms.numberToString",
+                            scale: "{that}.options.scale"
```

Comment:
Would be helpful to add code comments to explain the need of having implementers to provide `options.scale` and what it's for.